### PR TITLE
Remove reprecated flags from Builder

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

**Review by songy23 — COMMENTED** [review]
Some more references need to be cleaned up, e.g. https://github.com/open-telemetry/opentelemetry-collector/blob/37184b0b3fb2c789217e1801d56f3624a8d89721/Makefile#L160 

https://github.com/open-telemetry/opentelemetry-collector/blob/37184b0b3fb2c789217e1801d56f3624a8d89721/cmd/builder/test/test.sh#L36

**Comment by bogdandrutu** [comment]
@songy23 had to leave `--output-path` because of the test usage, will do a separate PR for that.

File: cmd/builder/internal/command.go [hunk 1]
```diff
@@ -83,35 +77,10 @@ configuration is provided, ocb will generate a default Collector.
...
-	cmd.Flags().StringVar(&cfg.Distribution.Go, distributionGoFlag, "", "The Go binary to use during the compilation phase. Default: go from the PATH")
-	if err := cmd.Flags().MarkDeprecated(distributionGoFlag, "use config distribution::go"); err != nil {
-		return nil, err
-	}
-	cmd.Flags().StringVar(&cfg.Distribution.Module, distributionModuleFlag, "go.opentelemetry.io/collector/cmd/builder", "The Go module for the new distribution")
-	if err := cmd.Flags().MarkDeprecated(distributionModuleFlag, "use config distribution::module"); err != nil {
-		return nil, err
-	}
-	// version of this binary
+	} // version of this binary
```
Comment (songy23) [inline cmd/builder/internal/command.go:83]:
```suggestion
	} 
	// version of this binary
```

**Review by songy23 — COMMENTED** [review]
This broke `make genotelcorecol` https://github.com/open-telemetry/opentelemetry-collector/actions/runs/11598880819/job/32295780346?pr=11576

File: cmd/builder/internal/command.go (hunk 1 above)
Comment (bogdandrutu) [reply cmd/builder/internal/command.go]: Done.

**Comment by bogdandrutu** [comment]
@songy23 everything works now.

**Review by songy23 — APPROVED** [review]

**Comment by andrzej-stencel** [comment]
The `--go` option is used in Releases repo: https://github.com/open-telemetry/opentelemetry-collector-releases/blob/4201470db503a8294ebce058c9ad46350957aadd/scripts/build.sh#L45.

**Pull request merged**